h2/src/main/org/h2/value/Transfer.java

```diff
@@ -494,6 +494,20 @@ public void writeValue(Value v) throws IOException {
             }
             break;
         }
+        case Value.ENUM: {
+            ValueEnum ve = (ValueEnum) v;
+            String[] enumerators = ve.getEnumerators();
+
+            writeInt(enumerators.length);
+
+            for (String member : enumerators) {
+                writeString(member);
+            }
+
+            writeInt(ve.getInt());
+
+            break;
+        }
```

Comment:
this code is used when communicating between client and server. In which case there is no need to send all of the enumerators. At most you need to send one integer value and one string value of the enumerator.